File: 242-valid-anagram/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        counter = dict()
        for c in s:
            counter[c] = counter.get(c, 0) + 1
        for c in t:
            if c in counter.keys() and counter[c] != 0:
                counter[c] -= 1
            else:
                return False
        for key, val in counter.items():
            if val != 0:
                return False
        return True

File: 242-valid-anagram/242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;
        HashMap<Character, Integer> freq = new HashMap<>();
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            freq.put(c, freq.getOrDefault(c, 0) + 1);
        }
        for (int j = 0; j < t.length(); j++) {
            char c = t.charAt(j);
            if (freq.containsKey(c)) {
                freq.put(c, freq.get(c) - 1);
            } else {
                return false;
            }
        }
        
        for (Character c : freq.keySet()) {
            if (freq.get(c) != 0) return false;
        }
        return true;
    }
}